server/src/main/java/org/apache/cloudstack/network/RoutedIpv4ManagerImpl.java

```diff
@@ -0,0 +1,1563 @@
+import java.util.stream.Collectors;
+
+
+public class RoutedIpv4ManagerImpl extends ComponentLifecycleBase implements RoutedIpv4Manager {
+
+    @Inject
+    DataCenterIpv4GuestSubnetDao dataCenterIpv4GuestSubnetDao;
+    @Inject
+    Ipv4GuestSubnetNetworkMapDao ipv4GuestSubnetNetworkMapDao;
+    @Inject
+    FirewallService firewallService;
+    @Inject
+    FirewallManager firewallManager;
+    @Inject
+    FirewallRulesDao firewallDao;
+    @Inject
+    NetworkServiceMapDao networkServiceMapDao;
+    @Inject
+    NetworkOfferingServiceMapDao networkOfferingServiceMapDao;
+    @Inject
+    NetworkOfferingDao networkOfferingDao;
+    @Inject
+    NetworkModel networkModel;
+    @Inject
+    AccountManager accountManager;
+    @Inject
+    VpcOfferingDao vpcOfferingDao;
+    @Inject
+    VpcOfferingServiceMapDao vpcOfferingServiceMapDao;
+    @Inject
+    VpcDao vpcDao;
+    @Inject
+    BgpPeerDao bgpPeerDao;
+    @Inject
+    BgpPeerDetailsDao bgpPeerDetailsDao;
+    @Inject
+    BgpPeerNetworkMapDao bgpPeerNetworkMapDao;
+    @Inject
+    NetworkDao networkDao;
+    @Inject
+    BGPService bgpService;
+
+    @Override
+    public String getConfigComponentName() {
+        return RoutedIpv4Manager.class.getSimpleName();
+    }
+
+    @Override
+    public ConfigKey<?>[] getConfigKeys() {
+        return new ConfigKey[] {
+                RoutedNetworkIPv4MaxCidrSize, RoutedNetworkIPv4MinCidrSize, RoutedIPv4NetworkCidrAutoAllocationEnabled,
+                RoutedVpcIPv4MaxCidrSize, RoutedVpcIPv4MinCidrSize
+        };
+    }
+
+    @Override
+    public List<Class<?>> getCommands() {
+        final List<Class<?>> cmdList = new ArrayList<Class<?>>();
+        cmdList.add(CreateIpv4GuestSubnetCmd.class);
+        cmdList.add(DeleteIpv4GuestSubnetCmd.class);
+        cmdList.add(ListIpv4GuestSubnetsCmd.class);
+        cmdList.add(UpdateIpv4GuestSubnetCmd.class);
+        cmdList.add(DedicateIpv4GuestSubnetCmd.class);
+        cmdList.add(ReleaseDedicatedIpv4GuestSubnetCmd.class);
+        cmdList.add(CreateIpv4SubnetForGuestNetworkCmd.class);
+        cmdList.add(ListIpv4SubnetsForGuestNetworkCmd.class);
+        cmdList.add(DeleteIpv4SubnetForGuestNetworkCmd.class);
+        cmdList.add(CreateRoutingFirewallRuleCmd.class);
+        cmdList.add(ListRoutingFirewallRulesCmd.class);
+        cmdList.add(UpdateRoutingFirewallRuleCmd.class);
+        cmdList.add(DeleteRoutingFirewallRuleCmd.class);
+        cmdList.add(CreateBgpPeerCmd.class);
+        cmdList.add(DeleteBgpPeerCmd.class);
+        cmdList.add(ListBgpPeersCmd.class);
+        cmdList.add(UpdateBgpPeerCmd.class);
+        cmdList.add(DedicateBgpPeerCmd.class);
+        cmdList.add(ReleaseDedicatedBgpPeerCmd.class);
+        cmdList.add(ChangeBgpPeersForNetworkCmd.class);
+        cmdList.add(ChangeBgpPeersForVpcCmd.class);
+        return cmdList;
+    }
+
+
+    @Override
+    public DataCenterIpv4GuestSubnet createDataCenterIpv4GuestSubnet(CreateIpv4GuestSubnetCmd cmd) {
```

Comment:
Do we need to have events for some of the API handlers in this class?

Reply:
added